contracts/Bridge/BaseXVSProxyOFT.sol

```diff
@@ -0,0 +1,283 @@
+     */
+    mapping(uint16 => uint256) public chainIdToLast24HourReceived;
+    /**
+     * @notice Timestamp when the last 24-hour window started from remote chain.
+     */
+    mapping(uint16 => uint256) public chainIdToLast24HourReceiveWindowStart;
+    /**
+     * @notice Address on which cap check and bound limit is not appicable.
+     */
+    mapping(address => bool) public whitelist;
+
+    /**
+     * @notice Emmited when address is added to whitelist.
+     */
+    event SetWhitelist(address indexed addr, bool isWhitelist);
+    /**
+     * @notice  Emitted when the maximum limit for a single transaction from local chain is modified.
+     */
+    event SetMaxSingleTransactionLimit(uint256 oldMaxLimit, uint256 newMaxLimit);
+    /**
+     * @notice Emitted when the maximum daily limit of transactions from local chain is modified.
+     */
+    event SetMaxDailyLimit(uint256 oldMaxLimit, uint256 newMaxLimit);
+    /**
+     * @notice Emitted when the maximum limit for a single receive transaction from remote chain is modified.
+     */
+    event SetMaxSingleReceiveTransactionLimit(uint256 oldMaxLimit, uint256 newMaxLimit);
+    /**
+     * @notice Emitted when the maximum daily limit for receiving transactions from remote chain is modified.
+     */
+    event SetMaxDailyReceiveLimit(uint256 oldMaxLimit, uint256 newMaxLimit);
+    /**
+     * @notice Event emitted when oracle is modified.
+     */
+    event OracleChanged(address indexed oldOracle, address indexed newOracle);
+
+    constructor(
+        address tokenAddress_,
+        uint8 sharedDecimals_,
+        address lzEndpoint_,
+        address oracle_
+    ) BaseOFTV2(sharedDecimals_, lzEndpoint_) {
+        innerToken = IERC20(tokenAddress_);
+
+        (bool success, bytes memory data) = tokenAddress_.staticcall(abi.encodeWithSignature("decimals()"));
+        require(success, "ProxyOFT: failed to get token decimals");
+        uint8 decimals = abi.decode(data, (uint8));
+
+        require(sharedDecimals_ <= decimals, "ProxyOFT: sharedDecimals must be <= decimals");
+        ld2sdRate = 10**(decimals - sharedDecimals_);
+
+        ensureNonzeroAddress(oracle_);
+        oracle = ResilientOracleInterface(oracle_);
+    }
+
+    /**
+     * @notice Set the address of the ResilientOracle contract.
+     * @dev Reverts if the new address is zero.
+     * @param oracleAddress_ The new address of the ResilientOracle contract.
+     */
+    function setOracle(address oracleAddress_) external onlyOwner {
+        ensureNonzeroAddress(oracleAddress_);
+        emit OracleChanged(address(oracle), oracleAddress_);
+        oracle = ResilientOracleInterface(oracleAddress_);
+    }
+
+    /**
+     * @notice Sets the limit of single transaction amount.
+     * @param chainId_ Destination chain id.
+     * @param limit_ Amount in USD.
+     */
+    function setMaxSingleTransactionLimit(uint16 chainId_, uint256 limit_) external onlyOwner {
```

Comment:
We need to refactor and store lz contracts in our repo in order to use ACM because the lz's parent contracts contain configuration setters that onlyowner can call.

Reply:
My fault, I missed the point of having the `XVSBridgeAdmin`. I understand the owner of the XVSProxyOFT's contracts will be a `XVSBridgeAdmin` contract, where the ACM is already integrated